desktop-widgets/configuredivecomputerdialog.cpp

```diff
@@ -1242,11 +1243,28 @@ void ConfigureDiveComputerDialog::reloadValuesSuuntoVyper()
 void ConfigureDiveComputerDialog::reloadValuesOSTC4()
 {
 	ui.serialNoLineEdit_4->setText(deviceDetails->serialNo);
-	ui.firmwareVersionLineEdit_4->setText(deviceDetails->firmwareVersion);
+	ui.firmwareVersionLineEdit_4->setText(deviceDetails->firmwareVersionString);
 	ui.customTextLlineEdit_4->setText(deviceDetails->customText);
 	ui.modelLineEdit_4->setText(deviceDetails->model);
 	ui.diveModeComboBox_4->setCurrentIndex(deviceDetails->diveMode);
 	ui.lastDecoSpinBox_4->setValue(deviceDetails->lastDeco);
+	if (deviceDetails->firmwareVersion >= 0x01060000) {
+		// 'Cave' brightness level was introduced in 1.6.0
+		ui.brightnessComboBox_4->setItemText(0, tr("Cave"));
+		ui.brightnessComboBox_4->setItemText(1, tr("Eco"));
+		ui.brightnessComboBox_4->setItemText(2, tr("Medium"));
+		ui.brightnessComboBox_4->setItemText(3, tr("High"));
+		(qobject_cast<QListView *>(ui.brightnessComboBox_4->view()))->setRowHidden(3, false);
+		QStandardItem* item = (qobject_cast<QStandardItemModel*>(ui.brightnessComboBox_4->model()))->item(3);
+		item->setFlags(item->flags() & Qt::ItemIsEnabled);
+	} else {
+		ui.brightnessComboBox_4->setItemText(0, tr("Eco"));
+		ui.brightnessComboBox_4->setItemText(1, tr("Medium"));
+		ui.brightnessComboBox_4->setItemText(2, tr("High"));
+		(qobject_cast<QListView *>(ui.brightnessComboBox_4->view()))->setRowHidden(3, true);
+		QStandardItem* item = (qobject_cast<QStandardItemModel*>(ui.brightnessComboBox_4->model()))->item(3);
+		item->setFlags(item->flags() & ~Qt::ItemIsEnabled);
+	}
```

Comment:
I haven't cared much about compatibility with anything except the last/later versions in any of these codebases. 1.6.0 is about 5 months old, so I hope most people have upgraded by now.

Btw. From what I see it the Cave setting has been there since 1.3.0.

So, I'd suggest the whole firmware version magic and just adding the Cave entry to the combo box,.